AU financial year and timezone

jonathaningram opened the pull request.

andrewoh531 commented on admin/app/main/views/dashboard.py:
```diff
@@ -402,8 +402,8 @@ def get_months_for_financial_year(year, time_format='%B'):
-            get_months_for_year(4, 13, year) +
-            get_months_for_year(1, 4, year + 1)
+            get_months_for_year(7, 13, year) +
+            get_months_for_year(1, 7, year + 1)
```
I think the financial year for say 2018 would be 2017-07-01 to 2018-06-30 not 2018-07-01 to 2019-06-30.

https://en.wikipedia.org/wiki/Fiscal_year#Australia
https://www.taxback.com/blog/australia-key-tax-dates/

jonathaningram replied:
You're right that 17/18 financial year has "year=2018". Does this affect this section of code specifically or the PR?

andrewoh531 commented on api/app/dao/date_util.py:
```diff
@@ -2,14 +2,14 @@
-            get_months_for_year(1, 4, year + 1)
+        convert_aest_to_utc(month) for month in (
+            get_months_for_year(7, 13, year) +
+            get_months_for_year(1, 7, year + 1)
```
As above regarding financial year date ranges.

Maybe this and the other method can be moved into `utils/notification_utils`

andrewoh531 commented on api/app/dao/date_util.py:
```diff
@@ -20,31 +20,32 @@ def get_months_for_year(start, end, year):
+     to UTC. The tzinfo is lastly removed from the datetime because the database
+     stores the timestamps without timezone.
+     :param year: the year for which to calculate the 1 July, 00:00 AEST
+     :return: the datetime of 1 July for the given year, for example 2016 = 2016-06-30 14:00:00
```
As above regarding financial year date ranges.

andrewoh531 commented on api/app/dao/services_dao.py:
```diff
@@ -320,9 +320,9 @@ def dao_fetch_monthly_historical_stats_for_service(service_id, year):
+            datetime(year, month, 1) for month in range(7, 13)
         ] + [
-            datetime(year + 1, month, 1) for month in range(1, 4)
+            datetime(year + 1, month, 1) for month in range(1, 7)
```
As above regarding financial year date ranges.

andrewoh531 commented on api/app/service/statistics.py:
```diff
@@ -18,9 +18,9 @@ def format_monthly_template_notification_stats(year, rows):
+            datetime(year, month, 1) for month in range(7, 13)
         ] + [
-            datetime(year + 1, month, 1) for month in range(1, 4)
+            datetime(year + 1, month, 1) for month in range(1, 7)
```
As above regarding financial year date ranges.

andrewoh531 commented on api/app/utils.py:
```diff
@@ -5,7 +5,7 @@
 from notifications_utils.template import SMSMessageTemplate, PlainTextEmailTemplate
 
-local_timezone = pytz.timezone("Europe/London")
+local_timezone = pytz.timezone("Australia/Sydney")
```
Should we move the `Australia/Sydney` timezone string to a common place like the `utils` folder?

andrewoh531 commented on api/app/utils.py:
```diff
@@ -48,30 +50,30 @@ def get_london_midnight_in_utc(date):
     return func.date_trunc(
         "month",
-        func.timezone("Europe/London", func.timezone("UTC", column))
+        func.timezone("Australia/Sydney", func.timezone("UTC", column))
```
Maybe use the `local_timezone` variable

jonathaningram commented:
@dagoof @andrewoh531 can you please review. Notes:

1. Added `make fmt` and removed the line length from lint because it's annoying and formatter should just fix it for me. Can revisit later.
2. Utils tests fail. Some are related to dates. I would imagine some are related to this, but haven't looked into them. I made a ticket for fixing utils tests, so I figure we can visit in that.
3. There are still some instances of "BST" and "Europe/London". There is a lot in this PR already, and there's lots to learn about the various billing and reporting, and I think some of the remaining instances are related to that (including a DB column named `bst_date` or something). Not sure that I want to go down that path in this PR.
4. Will squash.
